### PR TITLE
ADO: partial scan using extension storage

### PR DESCRIPTION
**Pull request by Garima-msft**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.AzureDevOps/Framework/Managers/PartialScanManager.ps1 [hunk 1]
```diff
@@ -66,13 +68,50 @@ class PartialScanManager
...
+
+					try {
+						$webRequestResult = Invoke-RestMethod -Uri $uri -Method Get -ContentType "application/json" -Headers @{Authorization=("Basic {0}" -f $base64AuthInfo)}
+						$this.ScanPendingForResources = $webRequestResult.value | ConvertFrom-Json
+                        $this.isRTFAlreadyAvailable = $true;
+						write-host "Get request ran successfully"
+					}
+					catch
+					{
+						write-host "error in Get request from extension storage :  $uri"
```
Comment (abhaydaga) [inline src/AzSK.AzureDevOps/Framework/Managers/PartialScanManager.ps1:106]: remove

File: src/AzSK.AzureDevOps/Framework/Managers/PartialScanManager.ps1 [hunk 2]
```diff
@@ -153,8 +186,39 @@ class PartialScanManager
...
+				
+				try {
+					if ($this.ResourceScanTrackerObj.ResourceMapTable -ne $null){
+						$webRequestResult = Invoke-WebRequest -Uri $uri -Method Delete -ContentType "application/json" -Headers @{Authorization = ("Basic {0}" -f $base64AuthInfo) } 
+						$this.ResourceScanTrackerObj = $null
+						write-host "Resource tracker removed successfully"
+					}
+				}
+				catch {
+					write-host "error in delete request from extension storage :  $uri"
```
Comment (abhaydaga) [inline src/AzSK.AzureDevOps/Framework/Managers/PartialScanManager.ps1:215]: remove

File: src/AzSK.AzureDevOps/Framework/Managers/PartialScanManager.ps1 [hunk 3]
```diff
@@ -205,15 +264,63 @@ class PartialScanManager
...
+                    }
+
+					try {
+						$webRequestResult = Invoke-WebRequest -Uri $uri -Method Put -ContentType "application/json" -Headers @{Authorization = ("Basic {0}" -f $base64AuthInfo) } -Body $body 
+						write-host "Resource tracker updated successfully"
+                        $this.isRTFAlreadyAvailable = $true;
+					}
+					catch
+					{
+						write-host "error in WriteToResourceTrackerFile method : $uri"
```
Comment (abhaydaga) [inline src/AzSK.AzureDevOps/Framework/Managers/PartialScanManager.ps1:318]: remove

Comment (abhaydaga) [reply src/AzSK.AzureDevOps/Framework/Managers/PartialScanManager.ps1:318]: Could not update the resource tracker file.

File: src/AzSK.AzureDevOps/Framework/Managers/PartialScanManager.ps1 (hunk 2 above)
Comment (Garima-msft) [reply src/AzSK.AzureDevOps/Framework/Managers/PartialScanManager.ps1]: Done

File: src/AzSK.AzureDevOps/Framework/Managers/PartialScanManager.ps1 (hunk 3 above)
Comment (Garima-msft) [reply src/AzSK.AzureDevOps/Framework/Managers/PartialScanManager.ps1]: Done

**Pull request merged**